pdns/dnsdist-snmp.cc

```diff
@@ -193,6 +194,27 @@ static void registerGauge64Stat(const char* name, const oid statOID[], size_t st
                                                               HANDLER_CAN_RONLY));
 }
 
+static void registerGauge64Stat(const char* name, const oid statOID[], size_t statOIDLength, std::atomic<uint64_t>* ptr)
+{
+  if (statOIDLength != OID_LENGTH(queriesOID)) {
+    errlog("Invalid OID for SNMP Gauge64 statistic %s", name);
+    return;
+  }
+
+  if (s_statsMap.find(statOID[statOIDLength - 1]) != s_statsMap.end()) {
+    errlog("OID for SNMP Gauge64 statistic %s has already been registered", name);
+    return;
+  }
+
+  s_statsMap[statOID[statOIDLength - 1]] = ptr;
```

Comment:
I don't see at all how this can work, since `handleGauge64Stats()` expects only `DNSDistStats::statfunction_t` objects. `(*boost::get<DNSDistStats::statfunction_t>(&it->second))(str)` should fail when `it->second` is a pointer to a `std::atomic<uint64_t>`.